### DISCUSSION TITLE
Create site on the same server as the panel.

### DISCUSSION THREAD

**Discussion by donniewr** [vitodeploy/vito#158] [Ideas]
Hi, I really like this project, but I would say the only inconvenience is that you need to have a minimum of two servers. I think it would be nice if we could also create sites on the same server as the panel. This would be similar to most PHP control panels in the market.

Thanks.

**Comment by saeedvaziry** [comment]
@donniewr I've thought about this a lot and there were previous conversations in this regard. Unfortunately its not safe at all. any wrong behavior on the websites can break Vito instance and then you might lose access to all your websites and the server.
For such case I always recommend booting up VitoDeploy on your local machine using docker and then using it to manage your remote servers.

**Comment by donniewr** [comment]
> @donniewr I've thought about this a lot and there were previous conversations in this regard. Unfortunately its not safe at all. any wrong behavior on the websites can break Vito instance and then you might lose access to all your websites and the server. For such case I always recommend booting up VitoDeploy on your local machine using docker and then using it to manage your remote servers.

I agree, and it makes sense when you manage multiple servers to have the panel separated from the servers. But I think we should still have the option to create sites on the same server. Perhaps, if we choose to use the same servers as VitoDeploy, the option to connect to other servers will be disabled. This will allow us to use the panel with just one server, like other panels.

**Reply by saeedvaziry** [reply]
@donniewr have you tried this with docker? I think it should be possible tho.
Boot up vito via docker and use the same server for sites

**Reply by donniewr** [reply]
I just tested it, and it works without any issues. Would you recommend using this method for production sites?

**Reply by saeedvaziry** [reply]
Generally, I don't recommend it at all. but as long as the websites and the operations you do on the website will not conflict with the docker and the port that vito uses it should be fine.
